libvast/src/logger.cpp

```diff
@@ -207,14 +213,43 @@ bool setup_spdlog(const vast::invocation& cmd_invocation,
   spdlog::init_thread_pool(queue_size, defaults::logger::logger_threads);
   std::vector<spdlog::sink_ptr> sinks;
   // Add console sink.
-  auto stderr_sink
-    = std::make_shared<spdlog::sinks::ansicolor_stderr_sink_mt>(log_color);
-  stderr_sink->set_level(vast_loglevel_to_spd(vast_console_verbosity));
+  std::string default_sink_type
+    = systemd::connected_to_journal() ? "journald" : "stderr";
+  auto sink_type
+    = caf::get_or(cfg_file, "vast.console-sink", default_sink_type);
+  auto console_sink = [&]() -> spdlog::sink_ptr {
+    if (sink_type == "stderr") {
+      auto stderr_sink
+        = std::make_shared<spdlog::sinks::ansicolor_stderr_sink_mt>(log_color);
+      return stderr_sink;
+    } else if (sink_type == "journald") {
+#ifndef VAST_ENABLE_JOURNALD_LOGGING
+      std::cerr << "Cannot use 'journald' sink, vast was built without systemd "
+                   "support\n";
+      return nullptr;
+#else
+      auto spdlog_sink = std::make_shared<spdlog::sinks::systemd_sink_mt>();
+      return std::static_pointer_cast<spdlog::sinks::sink>(spdlog_sink);
+#endif
+    } else if (sink_type == "syslog") {
+      auto syslog_sink = std::make_shared<spdlog::sinks::syslog_sink_mt>(
+        "vast", /*options = */ 0, LOG_USER, /*enable_formatting = */ true);
+      return std::static_pointer_cast<spdlog::sinks::sink>(syslog_sink);
+    } else {
+      std::cerr << "Illegal vast.console-sink value '" << sink_type << "', "
```

Comment:
```suggestion
      std::cerr << "illegal vast.console-sink value '" << sink_type << "', "
```

Reply:
Might also want to consider using `fmt::print(stderr, ...)` here.

Reply:
Probably a good idea, but there are already a few other instances of `std::cerr`-based logging in the file, so I'll keep it as-is for now.